Public/Src/Sandbox/MacOs/Interop/Posix/memory.c

```diff
@@ -42,15 +42,58 @@ int GetRamUsageInfo(RamUsageInfo *buffer, long bufferSize)
     return KERN_SUCCESS;
 }
 
+static uint64_t ProcessTreeWorkingSetSize(pid_t pid, const rlim_t max_proc_count, bool *success)
+{
+    uint64_t children_usage = 0;
+
+    pid_t pids[max_proc_count];
+    int child_count = proc_listchildpids(pid, pids, (int) max_proc_count);
+    *success &= child_count >= 0;
+
+    for (int i = 0; (i < child_count) && *success; i++)
+    {
+        int pid = pids[i];
+        rusage_info_current rusage;
+        *success &= proc_pid_rusage(pid, RUSAGE_INFO_CURRENT, (void **)&rusage) == 0;
+        children_usage += rusage.ri_resident_size;
+
+        pid_t childs_pids[max_proc_count];
+        int count = proc_listchildpids(pid, childs_pids, (int) max_proc_count);
```

Comment:
why do you need to call `proc_listchildpids` here again?  When you recursively call `ProcessTreeWorkingSetSize(pid, max_proc_count, success)` it will get called then and then if the count is 0 it will immediately return.